### ISSUE TITLE
Investigate stability of new HSapplication initialisers

### ISSUE THREAD

**Issue by cmsj** [Hammerspoon/hammerspoon#2360]
https://github.com/Hammerspoon/hammerspoon/commit/32cd3cb0af30080a0ae87b3ac7931c0ab2fd38e9 has left a worry about the new HSapplication code - that commit, while sensible, shouldn't have been necessary.

I need to figure out what's going on with the PID initialisers, before we can cut a release.

**Comment by latenitefilms** [comment]
FWIW, as previously discussed with @asmagill - I'm still seeing a lot of these errors:

```lua
2020-05-10 20:19:24: ********
2020-05-10 20:19:24: 20:19:24 ERROR:   LuaSkin: Unable to fetch NSRunningApplication for pid: 2696
2020-05-10 20:19:24: ********
```

**Comment by cmsj** [comment]
@latenitefilms ok that’s good, that means we have a canary for when I come up with a fix :)

**Comment by cmsj** [comment]
@latenitefilms just to confirm, you're still seeing that error, right? I've yet to provoke a failure here.

**Comment by cmsj** [comment]
@latenitefilms if you are still hitting it, could you do me a favour to try and figure out where it's coming from and add this:

`        @throw [NSException exceptionWithName:@"hs.application initWithPid" reason:@"Failed to find pid" userInfo:nil];`

after the `logError` on line 117 of `Hammerspoon/Main/HSuicore.m` - that should cause a Lua stack trace to be printed, which at least might help me figure out what CP is doing to trigger this.

**Comment by latenitefilms** [comment]
Sorry for the delayed reply - been flat out. Sadly, Melbourne's going back into lockdown for 6 weeks, so once I wrap up some current jobs, I should have a bit more free time to jump back on Hammerspoon/CommandPost. Will keep you posted.

**Comment by latenitefilms** [comment]
Apologies - I forgot about this - will add in this debugging code and let you know when it crashes.

**Comment by latenitefilms** [comment]
I think you can safely ignore the below errors I was seeing...

```lua
2020-05-10 20:19:24: ********
2020-05-10 20:19:24: 20:19:24 ERROR:   LuaSkin: Unable to fetch NSRunningApplication for pid: 2696
2020-05-10 20:19:24: ********
```

I think they're caused by the fact I'm using `hs.application.watcher` to detect changing applications, and sometimes when an application is in the process of closing, because CommandPost is doing a lot of stuff, by the time CommandPost gets around to processing code based on PID, the application is already closed.

Either way, I've had this code merged into CommandPost for three months now and haven't noticed any issues or problems.

**Comment by cmsj** [comment]
Ok, thanks
